pandas/tests/frame/indexing/test_setitem.py

```diff
@@ -339,3 +339,14 @@ def test_setitem_boolean_mask(self, mask_type, float_frame):
         expected = df.copy()
         expected.values[np.array(mask)] = np.nan
         tm.assert_frame_equal(result, expected)
+
+    def test_setitem_loc_only_false_indexer_dtype_changed(self):
+        # GH#37550
+        # Dtype is only changed when value to set is a Series
+        df = DataFrame({"a": ["a"], "b": [1], "c": [1]})
+        df.loc[[False], ["b"]] = 10 - df["c"]
```

Comment:
can you make clear either in the test name and/or comment that/whether the type of indexer for `["b"]` is relevant

Reply:
Added comment and parametrized test